### ISSUE TITLE
Intermediate CA are recognized as external servers

### ISSUE THREAD

**Issue by Hakujou** [opoplawski/ansible-pfsense#17]
Hello,

Currently if I add a CA and an intermediate CA right after it, the intermediate CA is recognized as an external certificate.
If I add it through the WebUI, it is properly recognized.
<img width="387" alt="Capture d’écran 2019-03-20 à 15 56 51" src="https://user-images.githubusercontent.com/1732743/54694624-cc4ec480-4b28-11e9-8462-69e6b17c1e16.png">
<img width="397" alt="Capture d’écran 2019-03-20 à 15 56 14" src="https://user-images.githubusercontent.com/1732743/54694625-cc4ec480-4b28-11e9-8d78-be607acad83c.png">

Is there a way to recognized the intermediate CA as intermediate ? Thanks

**Comment by opoplawski** [comment]
I suspect there is some code in pfsense to recognize that.  We'll probably need to handle it manually.  Can you post the correct intermediate CA config from the config.xml file?

**Comment by Hakujou** [comment]
For privacy purposes, I'd rather not disclose my entire config.xml, but here's the relevant part:
```
<ca>
                <crt>LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUZqRENDQTNTZ0F3SUJBZ0lKQU5oTWhmY2lZMTdwTUEwR0NTcUdTSWIzRFFFQkN3VUFNRk14Q3pBSkJnTlYKQkFZVEFrWlNNUTh3RFFZRFZRUUlEQVpHY21GdVkyVXhEakFNQmdOVkJBY01CVkJoY21sek1ROHdEUVlEVlFRSwpEQVpVYjNKeGRXVXhFakFRQmdOVkJBTU1DVlJ2Y25GMVpTQkRRVEFlRncweE9EQTRNVE14TXpFNU5EQmFGdzB6Ck9EQTRNRGd4TXpFNU5EQmFNRk14Q3pBSkJnTlZCQVlUQWtaU01ROHdEUVlEVlFRSURBWkdjbUZ1WTJVeERqQU0KQmdOVkJBY01CVkJoY21sek1ROHdEUVlEVlFRS0RBWlViM0p4ZFdVeEVqQVFCZ05WQkFNTUNWUnZjbkYxWlNCRApRVENDQWlJd0RRWUpLb1pJaHZjTkFRRUJCUUFEZ2dJUEFEQ0NBZ29DZ2dJQkFPSEtEdjRSTG10cENoc1lUOStvCmRBYnE4dDdtVEx6bGh2aVQwaUVhQ01uNEZlZDlEdG5EbGpORUNUWXNhc2l2eVVYN3VIZ3VVYjdSQ1RPU0QvSlEKUUUvMmtRb1FhNTRUbnhKcm5IVng5cWY4VEliQWVwYlFZVjJFMW9ZOE9RZWhvV3F4OHJ3TDF2U3JZQVJYYjdIaApVQzlUWVdRL3ZSOFRUby8zS3hlOFFHOTNlOGQzVmJ4VXcyUWdSYWc5YjFKRXo2dm5lMVNXU21TWFJ0d2dHK2d0CklIU0Q3dXRJQUYzYnFnRko3MnRhYXdoejRCT1MxdUVzYnJ6QVpCY0xzS1VRanpLWHpoeDUzMHhMOGNPZUhKVkoKckVDaFVUc1lCd0xtWVRNcG0zTUNycEtWQUZrU3doSXVPd0Voako5UmlwcHJKSlhnUEJtbG5sVWFLRXFra0lQQgo2ZER5TXF4WWN0c0J1OWVrT3BQb1ZZM3kvT3VvRDRVa3MxeWczcGkvV0EwRnpZdnpRUThtWWtQMGtncWdINTVFClhXYkRMcGV2WWJvQjV2eUJCbE1zM2VWeDVEbHVvdkxHYlF2aWNaVURXTnFxTlVuS3J3NVQ2UGZWMjdaTDRHWHIKbkUzQmRKUkZiSFJKMXBKNDRhQ1Z0dm1UNnVLUithVkdoVEliWnl1ZXpJenRsM01WMTlmcWhJWTV3SjRIOXZ3bApzT3hWdTV5SFU1dXFTMGlVNWdBUU4rVGgyT0N2Wm1RdFhrOXZjMUZKdEFqcG9pVVhhRXFFem1sTmhNRGxoYnM5CjUyV0F2bFFzelZGQWY4K3YwbHJmbDZTTFBTazloNUErRWR1WnlTd29VS3h3RUtidkNpNEhqZEppUFk4YW5NSnUKNlptbVNHUlJBNnhsOU1BbkxTV2tQYkpqQWdNQkFBR2pZekJoTUIwR0ExVWREZ1FXQkJRN2piaDBUekpTSU1zNwpHTlNwK1VqcllPeUpsekFmQmdOVkhTTUVHREFXZ0JRN2piaDBUekpTSU1zN0dOU3ArVWpyWU95Smx6QVBCZ05WCkhSTUJBZjhFQlRBREFRSC9NQTRHQTFVZER3RUIvd1FFQXdJQmhqQU5CZ2txaGtpRzl3MEJBUXNGQUFPQ0FnRUEKWGpaNHUzR1lINWRRMEFOb0NrREkxQ2FZbWEyRVkvZnBhcXZja1hqckh2QkhFdFJnbTdkZEYyL2p2YzlvWFNSSQpNQ1hjakdrR2tsSVhIQk0zV0VQN0prQVBTUjlMUHpiUjZCb1Z1bktGWDJOMndZbFlaOTRNZmtjeGFldjQ3OEpjCjVUZm5aaHVSWHRaSzViTjB1RkgvVzZDUmZ2SVBlU1JKWEE4c205MzFyTDkvd0lhQlZXSVBZNTZuTTdSaW5HZS8KOXhzdUsxcVR3VFFhUDQ0L0NCamZFZjlUT2N4bFBFanA1MHMrTUpMcTlOZTFZeURxWjA2VnhJMU1QNy9rSVVGWApSaGc0TjZid1p1dFVLWlV1eHd1bERlV3Yxc21UQnQ2THNvMW5DdzZOK0tDSytGbGFvRHF4bmpsOUZvN0F1N0l0CnFac3ZWcTVLdW00MFBqbmF4cUNmZFZwWjdUYWVKbDJmL1ZRWnFGbUVWdmFSbDhQbTB3d05VSVcvUGJKMU55RXYKRkNpakFvNTQxZ1F1amtueEdONStMU2RjTG14KzRVTnJ1aGFTdUJFZ0NkZElkQkJHczZGV0lmRjZSZnorRE82aApkZGhnZTV2TjY2VUZDNFhFek5xdHB4bGYzVVFGOU9UL0U3WCtUMFpEYXRpblQ5Q3FwWGZZWHoyakNVdU15ejdlCi9ZM3ZDUWNUeXRMV1FnalQ3ZlJrN2VvUmVmczNtMUg2ZHZiTGNVaHBJMTZZWll1UkNwa2NwOXV1Mms2dHBiamUKRDNWYmdybkFCcnZOMmV3NUFVdlRTZ1ZqeXFGdmQ3Qm9pYm10NXhzVDViK1VTUVloZmFqUWFxY2lISzdPSWoyagpWT1BIL0FidG1peUZwWVZnQ1BJNEdISDJSdlVZaVVFM3NpVmNFME8wY1VvPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==</crt>
                <descr>Torque CA</descr>
                <refid>5c923db450a2c</refid>
        </ca>
        <ca>
                <refid>5c9254fd9fb44</refid>
                <descr>Torque Intermediate CA</descr>
                <crt>LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tDQpNSUlGaFRDQ0EyMmdBd0lCQWdJQ0VBQXdEUVlKS29aSWh2Y05BUUVMQlFBd1V6RUxNQWtHQTFVRUJoTUNSbEl4DQpEekFOQmdOVkJBZ01Ca1p5WVc1alpURU9NQXdHQTFVRUJ3d0ZVR0Z5YVhNeER6QU5CZ05WQkFvTUJsUnZjbkYxDQpaVEVTTUJBR0ExVUVBd3dKVkc5eWNYVmxJRU5CTUI0WERURTRNRGd4TXpFek1qTXdNbG9YRFRJNE1EZ3hNREV6DQpNak13TWxvd1VERUxNQWtHQTFVRUJoTUNSbEl4RHpBTkJnTlZCQWdNQmtaeVlXNWpaVEVQTUEwR0ExVUVDZ3dHDQpWRzl5Y1hWbE1SOHdIUVlEVlFRRERCWlViM0p4ZFdVZ1NXNTBaWEp0WldScFlYUmxJRU5CTUlJQ0lqQU5CZ2txDQpoa2lHOXcwQkFRRUZBQU9DQWc4QU1JSUNDZ0tDQWdFQXdQNWllME14aWNLMStIU3J3aWVvZHl5VnpLaFZnNXBIDQp2YnlPbFNLR1h6dDRMNlBRcFlSUEwvVGpCd3pTUkxheVZ2VmhObjJFeGNob3B1blFjNFJCYi9pcWhnMGVNZnF0DQpocjJ5YXlDT25YYTNpb0RadWQ3c25vNlE0cER3S2VobGp1Y2VwOWs5ZGJEWXJsVXIwbk85VTZkQVRxbkRsMXUzDQpKN016L3FPZVI5MldhbHdlV3hteGxjdjJKc0JJNHJmdkg0M2J0YXdFQ1IyNmQ4bTZmekJDNGdHWVlVSW03OVcyDQoyMzNOZkdWbVdYSTJrdmZtWUZaazRRQTg0d1hDdHNKNC85RG54MVZEeE0ydW9CY1RwY2Fub1IxUWI1bGNzcC9PDQo4TzZ0Tm91aDFyYXpSbkhyUHRwMUV5UW51dU4xVkFON2lBSGxZYWRzYXdCcm5zaFl2TXUrY3g3S0Y2OFhQRmx6DQpieENmRmNFQWRQZ2h1SUJoUzVIeTVQNm9nK2NCV1R2dDRkdTd3L09ZNzU5R2FFN0o1QTBVQ2U2bVNpUUxMWFZvDQpoQ1N5bUJVam90c2ZERTcxNThRWmRnTCtoT3IzT3BwTWFnNSsxbmRmY0p4dzNEb1hPRmN1NjZMU2pYTFZSaEg5DQorUTFDcHVIUVRqN2NNT1BwT3U4aVM0T3Z0SXEyL0tLeTJkRTFnalRZU0h3bXhDZjNyVysxMXNTSmJzUmZZNEh6DQpUZm1mRk1ZUzU1WjZYWjVhM0gyK2thOTJWZzJRckFyQU9lclV3UUlQTVBDeWpxUll6R3B6QUFnUmwvQ21FMnFXDQpvVVlHUUpGVVdFTVpoV05OZkd4VHFSTGQ1ZkI4cmxQZzVlSnF4eXpCQXJyVXhiWEpicEYvanZxTTFJMmU3YUtPDQpISFhKd21XZmRCTUNBd0VBQWFObU1HUXdIUVlEVlIwT0JCWUVGUDN6c3QzVlNnV2lybkNvQmZkbEhCUDlwWTVyDQpNQjhHQTFVZEl3UVlNQmFBRkR1TnVIUlBNbElneXpzWTFLbjVTT3RnN0ltWE1CSUdBMVVkRXdFQi93UUlNQVlCDQpBZjhDQVFBd0RnWURWUjBQQVFIL0JBUURBZ0dHTUEwR0NTcUdTSWIzRFFFQkN3VUFBNElDQVFERDRnZkFDdmtGDQo1djlzUDBPc0lNeVV4NGZzWm1SSXVWcUpPdTRRdWwrekI1RURDemw5SHN4MTZ3RkNlVldwdCtKMlk4aisydzErDQpLL3U5cVpqZ0NmSUZJQlBnYm9INDlRVVh3eUVnUVZGMlU3aWo0RHZ6ei8vTWU4N3FUQ0puVTgzSVc2UFh6MU0wDQpZSEFCYjcwZTZjV09oZUluYk5EYlludnZueENHTi80SGZFaVlUZ3ZxemliVzMzaHJORUZrMW5ONDhlYnhOL2E2DQplL0JIRTBsTnB6VUF3am1ZNDFXWDdGUWQ2a2EvZUxXQXhXenFCb2oxSWdJdmRnbE1MMmcwaFF6RlZjRWdpSzVODQowSnFWTEZFSGQzQjNDQlFTQ25iTkFpbUNNNXQyWlZEQUFWL25QS1JMdG83RnVuMmdWdkpHcHNuVzZXaHdFaGcvDQovRVZzanNuRUN4MkxCUzVjMEJ2YkZMV2xCN2thS0JZWWlLZC8wSjRpbncyUnRsa1Awbk5UVC9ZWkttMXMxSitEDQpUVG1PbGFaeVJTVWVVM1gwbnRSY1gyYmpkY1pxcmRmY3k0ckVwMEszTDJ0VnlXOTYyam16T01BN3ZnNVhkMXFzDQp6WVlJa3NGU1RvZzNBTjRmTHBPOGdJMWVvaVhxbk9vK3k0V2pWWThuaWFUbDkzL1ZXZkxGSGh2ZGxEcDU1a05yDQo0akJzYmNuYXZFUjNoZUI3RWFQcDdwZTQ1U1lvSHdIQVFJbDIwVXFwZ3k4NVZVdXFvNy8wWlVGTUllMzU2M1ZYDQpSTVA2cm96b1diRnpiakZZL1pZNnhKbmpyOE1wSStNeml3VnBJK3ZXdXI0NVVWc1YydjFrdThUNk1STjlWUWZvDQp6aUF5Y0lhQjhpRThocWcxU1g0M3V1V3I4c29PVDBLdjlBPT0NCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0=</crt>
                <serial>0</serial>
                <caref>5c923db450a2c</caref>
        </ca>
```

As far as I can see, there's a "<caref>" field pointing to the main CA.

**Comment by opoplawski** [comment]
This should be fixed with 664828054bccee0f9acecef849ab497424e21d7c
